team-builder/src/components/Form.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+const Form = ({ teamMember, teamList, name, email, role, handleNameChange, handleEmailChange, handleRoleChange, handleSubmit }) => {
+  return (
+    <div className="App">
+      <form onSubmit={event => handleSubmit(event)}>
+        <label>
+          Username:
+          <input type='text' value={name} placeholder={'name'} onChange={event => handleNameChange(event)} />
```

Comment:
This is why it was updating as you typed: you have the values set as name, email, and role instead of teamMember.name, teamMember.email, teamMember.role